subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -542,6 +542,44 @@ static int engine_add_observer(struct lwm2m_message *msg,
 	return 0;
 }
 
+int lwm2m_engine_update_period_min_observer(uint32_t obj_id, uint32_t res_id, uint32_t period_s)
+{
+	int i;
+
+	for (i = 0; i < CONFIG_LWM2M_ENGINE_MAX_OBSERVER; i++) {
+		if (observe_node_data[i].path.obj_id == obj_id &&
+			observe_node_data[i].path.res_id == res_id) {
+			observe_node_data[i].min_period_sec = period_s;
+			break;
+		}
+	}
+
+	if (i == CONFIG_LWM2M_ENGINE_MAX_OBSERVER) {
+		return -ENOMEM;
```

Comment:
I'd rather see `-ENOENT` here, `ENOMEM` is rather used if we run out of buffers / fail to allocate some memory. Same in other functions.

Reply:
Solved.